[IMP] Amazon_connector: manage errors in the pickings (#3193)
odoo/documentation: content/applications/sales/sales/amazon_connector/manage.rst

.. seealso::
   - :doc:`../../../inventory_and_mrp/inventory/shipping/setup/third_party_shipper`

.. _manage/manage_delivery_errors:

Manage errors when synchronizing deliveries
-------------------------------------------

> Sometimes, Amazon can fail to correctly process all the information sent by Odoo. In this case, Odoo

Reviewer: Ah, you're right. I can't count today 😅

> Sometimes, Amazon can fail to correctly process all the information sent by Odoo. In this case, Odoo

Author: No proble problem at all! If I may give you an advice, for your next reviews, you can add a visual ruler on your IDE after the 100th character, it's helpful for this kind of things and will spare you a lot of time. =]

> Sometimes, Amazon can fail to correctly process all the information sent by Odoo. In this case, Odoo

Reviewer: Yea we use [Highlight](https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-highlight), but the regex we use in the extension to count characters was off by one 😅 Should be all fixed now.

sends an email to the Odoo user listing all the shipments that failed and the errors Amazon sent
with them. In addition, these shipments are flagged with a :guilabel:`Synchronization with Amazon
failed` tag.

Usually, the error can be corrected directly in the Amazon backend or in Odoo. If the problem is
corrected in Odoo, synchronize the shipment again using the :guilabel:`Retry Amazon Sync` button.

.. note::
   It might happen that Odoo receives a notification from Amazon saying that some delivery
   information was not processed, but without specifying which shipments were affected. In that
   case, all the shipments in an unknown state will be treated as if they failed to synchronize.
   Once Odoo receives a notification from Amazon saying that a shipment was processed, its tag will

>    change to :guilabel:`Synchronized with Amazon`. To speed up this process, on your Amazon account,

Reviewer: Break at or before 100th characters
```suggestion
   change to :guilabel:`Synchronized with Amazon`. To speed up this process, on your Amazon
```

Author: Same as the other comment =]

Reviewer: Right here too...

.. seealso::
   - :doc:`../../../inventory_and_mrp/inventory/shipping/setup/third_party_shipper`

.. _manage/manage_delivery_errors:

> Manage errors when synchronizing the deliveries

Reviewer 2: Deleting an unnecessary word
```suggestion
Manage errors when synchronizing deliveries
```

-----------------------------------------------

> When Odoo verifies that all the information sent to Amazon are correctly processed, it can happen
> that Amazon failed to do so. In that case, Odoo send an email listing all the shipments that failed

Reviewer 2: Editing this section for clarity and wordiness. I assume that Odoo sends an email to the Odoo database user in this case, but please correct me if I'm wrong.
```suggestion
Sometimes, Amazon can fail to correctly process all the information sent by Odoo. In this case, Odoo sends an email to the Odoo user listing all the shipments that failed
```

> and the errors Amazon sent with it. In addition, these shipment will be flagged with a

Reviewer 2: Editing for grammar and present tense
```suggestion
and the errors Amazon sent with them. In addition, these shipments are flagged with a
```

:guilabel:`Synchronization with Amazon failed` tag.

> Usually, the error will be clear enough for you to correct the problem, so you either manage this
> order directly in Amazon backend, or you correct the problem in Odoo and synchronize the shipment

Reviewer 2: Editing for clarity and wordiness. We also want to avoid using second-person language ("you") in order to make the writing sound more neutral and direct.
```suggestion
Usually, the error can be corrected directly in the Amazon backend or in Odoo. If the problem is corrected in Odoo, synchronize the shipment
```

again using the :guilabel:`Retry Amazon Sync` button.

.. note::

>    It might happen that Odoo receive as notification from Amazon saying that some delivery

Reviewer 2: Editing for grammar
```suggestion
   It might happen that Odoo receives a notification from Amazon saying that some delivery
```

>    information were not processed, but without saying which ones. In that case, all the shipments

Reviewer 2: Editing for grammar and clarity
```suggestion
   information was not processed, but without specifying which shipments. In that case, all the shipments
```

>    in an unknown state will be treated as if they failed to be synchronized. Once Odoo get a

Reviewer 2: Editing for wording
```suggestion
   in an unknown state will be treated as if they failed to synchronize. Once Odoo receives a
```

>    notification from Amazon saying that a shipment was processed, its tag will changed to

Reviewer 2: Editing for grammar
```suggestion
   notification from Amazon saying that a shipment was processed, its tag will change to
```

>    :guilabel:`Synchronized with Amazon`. You can speed up this information either by forcing the
>    synchronization of all your orders sooner, or by asking to :guilabel:`Recover Order` and entering
>    the concerned Amazon Order Reference.

Reviewer 2: Editing for clarity and deleting instances of "you". I don't have access to Amazon test credentials to test this, but please add the correct button name where I wrote "[insert button name]" :)
```suggestion
   :guilabel:`Synchronized with Amazon`. To speed up this process, click on :guilabel:`[insert button name]` to manually synchronize these orders, or click on :guilabel:`Recover Order` and enter the relevant Amazon Order Reference.
```

Author: I added the precision that the 2nd and 3rd buttons are on the Amazon Account, I hope it's clearer that way?

Reviewer 2: Very good, thank you!
